File: docs/competitors/python/factorial.py
def calculate_factorial(number):
    if number < 0:
        return -1
    if number == 0 or number == 1:
        return 1
    return number * calculate_factorial(number - 1)

try:
    number = int(input('Введите неотрицательное число: '))
    result = calculate_factorial(number)
    if result == -1:
        print("Факториал не определяется для отрицательных чисел")
    else:
        print(f'Результат: {result}')
except ValueError:
    print('Ошибка: введите целое число!')

File: docs/competitors/pascal/factorial.pas
program FactorialCalculator;
var
  number: integer;
  result_fact: integer;

function calculateFactorial(n: integer):integer;
begin
  if n < 0 then 
    calculateFactorial := -1
  else if ((n = 0) or (n = 1)) then 
    calculateFactorial := 1
  else 
    calculateFactorial := n * calculateFactorial(n - 1);
end;

begin
  Write('Введите неотрицательное число: ');
  ReadLn(number);
  
  result_fact := calculateFactorial(number);

  if result_fact = -1 then 
    WriteLn('Факториал не определяется для отрицательных чисел')
  else
    WriteLn('Результат: ', result_fact);
end.
